src/storage/src/hummock/state_store.rs

```diff
@@ -193,41 +194,54 @@ impl StateStore for HummockStorage {
     /// failed due to other non-EOF errors.
     fn get<'a>(&'a self, key: &'a [u8], epoch: u64) -> Self::GetFuture<'_> {
         async move {
-            let version = self.local_version_manager.get_version()?;
+            let (version, shared_buffer) = self
+                .local_version_manager
+                .get_version_and_shared_buffer(&(key..=key), epoch, false)?;
+
             // check epoch validity
             validate_epoch(version.safe_epoch(), epoch)?;
 
             // Query shared buffer. Return the value without iterating SSTs if found
-            if version.max_committed_epoch() < epoch {
-                if let Some(v) = self
-                    .shared_buffer_manager
-                    .get(key, (version.max_committed_epoch() + 1)..=epoch)
-                {
-                    self.stats.get_shared_buffer_hit_counts.inc();
-                    return Ok(v.into_user_value().map(|v| v.into()));
+            for batch in shared_buffer {
+                if batch.start_user_key() > key {
```

Comment:
Can we also filter with end_user_key()?

Reply:
get_overlap_batches already do the filtering. Actually, there is also no need to check start user key here. Fixedd.